readers/postgres/messages.go

```diff
@@ -141,9 +142,15 @@ func fmtCondition(chanID string, rpm readers.PageMetadata) string {
 		case "vb":
 			condition = fmt.Sprintf(`%s AND bool_value = :bool_value`, condition)
 		case "vs":
-			condition = fmt.Sprintf(`%s AND string_value = :string_value`, condition)
+			comparator := readers.ParseValueComparator(query)
```

Comment:
Instead of `if else` , we can try to `switch` case 
And also for comparator `=` we can use `string_value = :string_value `

Reply:
```suggestion
			comparator := readers.ParseValueComparator(query)
```

Reply:
We can use switch case like these all the comparators `<`, `<= `, `>` ,`>=`, `=` like given below
```suggestion
			switch comparator {
			case "=":
				condition = fmt.Sprintf("%s AND string_value = :string_value ", condition)
			case "<":
				condition = fmt.Sprintf("%s AND string_value LIKE '%%' || :string_value || '%%' AND string_value <> :string_value", condition)
			case "<=":
				condition = fmt.Sprintf("%s AND string_value LIKE '%%' || :string_value || '%%'", condition)
			case ">=":
				condition = fmt.Sprintf("%s AND :string_value LIKE '%%' || string_value || '%%'", condition)
			case ">":
				condition = fmt.Sprintf("%s AND :string_value LIKE '%%' || string_value || '%%' AND string_value <> :string_value", condition)
			}
```

Reply:
All the tests are failing after this change @arvindh123 .
I'm trying to debug the problem